src/core/valid/valid.rs

```diff
@@ -157,6 +157,16 @@ impl<A, E> Valid<A, E> {
     }
 }
 
+impl<A: std::fmt::Debug, E: std::fmt::Debug> Valid<A, E> {
+    pub fn unwrap(self) -> A {
+        self.0.unwrap()
+    }
+
+    pub fn unwrap_err(self) -> ValidationError<E> {
+        self.0.unwrap_err()
+    }
+}
+
```

Comment:
can be moved to `test` config. Unwrap isn't a feature that Valid need to offer.

Reply:
since it easily could be done with `valid.to_result().unwrap()` anyway, is there a point of hiding it?

Reply:
We need to ask this question —
1. Is Valid a wrapper on Result?
2. Is Valid is a value that allows horizontal composition of errors?

If we consider definition 1, then yes, exposing the common Result methods on top of Valid makes sense. However if we consider that Valid is a type of its own and usage of Result type internally is merely an implementation detail, then exposing unwrap doesn't seem right. For eg: Let us change the implementation of Valid to not use a Result type but something custom, in that case we will have an overhead of supporting this API.